Neos.Flow/Classes/Mvc/Routing/Dto/ActionUriSpecification.php

```diff
@@ -0,0 +1,169 @@
+<?php
+declare(strict_types=1);
+namespace Neos\Flow\Mvc\Routing\Dto;
+
+/*
+ * This file is part of the Neos.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\Mvc\ActionRequest;
+use Neos\Utility\Arrays;
+use Psr\Http\Message\UriInterface;
+
+/**
+ * @Flow\Proxy(false)
+ */
+final class ActionUriSpecification
+{
+    private function __construct(
+        private string $actionName,
+        private string $controllerName,
+        private string $packageKey,
+        /** @deprecated with Flow 8.3 - The use of subpackage keys is discouraged and only supported for backwards compatibility */
+        private ?string $subpackageKey,
+        private string $format,
+        private array $routingArguments,
+        private array $queryParameters,
+    ) {
+    }
+
+    public static function create(string $packageKey, string $controllerName, string $actionName): self
+    {
+        return new self($actionName, $controllerName, $packageKey, null, '', [], []);
+    }
+
+    public static function fromActionRequest(ActionRequest $request): self
+    {
+        return new self(
+            $request->getControllerActionName(),
+            $request->getControllerName(),
+            $request->getControllerPackageKey(),
+            $request->getControllerSubpackageKey(),
+            $request->getFormat(),
+            [],
+            []
+        );
+    }
+
+    /**
+     * @deprecated with Flow 8.3 - The use of subpackage keys is discouraged and only supported for backwards compatibility
+     */
+    public function withSubpackageKey(string $subpackageKey): self
```

Comment:
@bwaidelich is it intentional that we dont have `ActionUriSpecification::withPackageKey`

Reply:
Not really IIRC. Do you think we need it?

Reply:
i.e. the `fromActionRequest` constructor is meant to easily create URLs to actions of the same package.
To link to _other_ packages you can always use `create()` because the current action request is irrelevant for that. I guess that was my thinking behind omitting it

Reply:
i thought rewriting redirect like:
```
$targetActionUriSpecification = ActionUriSpecification::fromActionRequest(
    $this->request,
)
    ->withActionName($actionName)
    ->withRoutingArguments($arguments);


if ($packageKey !== null && str_contains($packageKey, '\\')) {
    [$packageKey, $subpackageKey] = explode('\\', $packageKey, 2);
} else {
    $subpackageKey = null;
}

if ($packageKey !== null) {
    $targetActionUriSpecification = $targetActionUriSpecification->withPackageKey($subpackageKey);
}
if ($subpackageKey !== null) {
    $targetActionUriSpecification = $targetActionUriSpecification->withSubpackageKey($subpackageKey);
}
```

Reply:
you might as well do these decisions beforehand and use create?

Reply:
I would encourage (by example) to avoid using the current request to build uris as much as possible.

Reply:
This is the way